Android init script test issue

What's the point of the first test? Or better yet, why does it always fail although the device works ok?

```
dnscrypt-proxy \
  --resolver-name="$RESOLVER_NAME" \
  --resolvers-list=/system/etc/dnscrypt-proxy/dnscrypt-resolvers.csv \
  --test=3600 && \
dnscrypt-proxy \
  --daemonize \
...
```

`[ERROR] Unable to retrieve server certificates` (no matter what server I choose) hence `dnscrypt-proxy` will never start.
Commenting that first command let's it start ok though.

--- a/dist-build/android-files/system/etc/init.d/99dnscrypt
+++ b/dist-build/android-files/system/etc/init.d/99dnscrypt
@@ -1,15 +1,34 @@
 #!/system/bin/sh
 
-RESOLVER_NAME=dnscrypt.org-fr
+while true
+do
+#check first server
+        RESOLVER_NAME=dnscrypt.org-fr
+        dnscrypt-proxy \
+        --resolver-name="$RESOLVER_NAME" \
+        --resolvers-list=/system/etc/dnscrypt-proxy/dnscrypt-resolvers.csv \
+        --test=3600
+        case "$?" in
+                0 ) break;;
+        esac
+        sleep 1
+
+# check second server (servers go down pretty often)
+        RESOLVER_NAME=okturtles
+        dnscrypt-proxy \
+        --resolver-name="$RESOLVER_NAME" \
+        --resolvers-list=/system/etc/dnscrypt-proxy/dnscrypt-resolvers.csv \
+        --test=3600
+        case "$?" in
+                0 ) break;;
+        esac
+        sleep 1
+done
 
 dnscrypt-proxy \
-  --resolver-name="$RESOLVER_NAME" \
-  --resolvers-list=/system/etc/dnscrypt-proxy/dnscrypt-resolvers.csv \
-  --test=3600 && \
-dnscrypt-proxy \
-  --daemonize \
-  --loglevel=3 \
-  --resolver-name="$RESOLVER_NAME" \
-  --resolvers-list=/system/etc/dnscrypt-proxy/dnscrypt-resolvers.csv && \
+--daemonize \
+--loglevel=3 \
+--resolver-name="$RESOLVER_NAME" \
+--resolvers-list=/system/etc/dnscrypt-proxy/dnscrypt-resolvers.csv && \
 iptables -t nat -A OUTPUT -p udp --dport 53 -j DNAT --to-destination 127.0.0.1 && \
 iptables -t nat -A OUTPUT -p tcp --dport 53 -j DNAT --to-destination 127.0.0.1

